src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -126,7 +126,8 @@ internal static bool IsDictionary(CSharpType type)
         {
             return type.IsFrameworkType &&
                    (type.FrameworkType == typeof(IDictionary<,>) ||
-                   type.FrameworkType == typeof(IReadOnlyDictionary<,>));
+                   type.FrameworkType == typeof(IReadOnlyDictionary<,>) ||
+                   type.FrameworkType.IsGenericType && type.FrameworkType.GetGenericTypeDefinition() == typeof(IDictionary<,>));
```

Comment:
The `type.FrameworkType` should already be the generic definition.

Reply:
In the case of SystemObjectType it is not

Reply:
then we should make it so. That's the contract of CSharpType.